documentation/modules/ROOT/pages/configuration/storage.adoc

```diff
@@ -0,0 +1,705 @@
+
+
+== File
+
+It's possible to persist the position (offsets) of the connector in a local file on disk. This allows Debezium to resume from where it left off after a restart. It's simple, fast, and ideal for single-node applications or testing scenarios.
+
+=== Offset Store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[offset-storage-file]]<<offset-storage-file,`offset.storage`>>
+|No default
+|Must be set to `org.apache.kafka.connect.storage.FileOffsetBackingStore`
+
+|[[offset-storage-file-filename]]<<offset-storage-file-filename,`offset.storage.file.filename`>>
+|No default
+|Path to the file where Debezium stores source offsets.
+
+|[[offset-flush-interval-ms]]<<offset-flush-interval-ms,`offset.flush.interval.ms`>>
+|6000ms
+|Interval in milliseconds to flush the current offset state to the file.
+
+|===
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[schema-history-internal-file]]<<schema-history-internal-file,`schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.file.history.FileSchemaHistory`
+|[[schema-history-internal-file-file]]<<schema-history-internal-file-file,`schema.history.internal.file`>>
+|No default
+|The path to the file that will be used to record the database schema history`
+
+|===
+
+== Memory
+
+`MemoryOffsetBackingStore` is a volatile, in-memory store used by Debezium Embedded to track source offsets. It keeps the offset state only during the application's runtime and loses it on shutdown or crash. It's ideal for testing or short-lived tasks but not suitable for production where offset persistence is needed.
+
+=== Offset Store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[offset-storage-memory]]<<offset-storage-memory,`offset.storage`>>
+|No default
+|Must be set to `org.apache.kafka.connect.storage.MemoryOffsetBackingStore`
+
+|===
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[schema-history-internal-memory]]<<schema-history-internal-memory,`schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.relational.history.MemorySchemaHistory`
+
+|===
+
+
+== JDBC
+The store uses arbitrary relational database to store the data for which an appropriate link:https://en.wikipedia.org/wiki/JDBC_driver[JDBC driver] is provided.
+Either the source database can be used as the storage or any other database.
+
+The users either use the default pre-configured DML and DDL statements or can override them with their own either for database dialect compatibility or to tailor them to their use case.
+
+
+=== Offset Store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[jdbc-offset-type]]<<jdbc-offset-type, `offset.storage`>>
+|No default
+|Must be set to `io.debezium.storage.jdbc.offset.JdbcOffsetBackingStore`.
+
```

Comment:
If this and similar properties have required values, can they be coded to carry the appropriate defaults?